protocol/daemons/bridge/client/client.go

```diff
@@ -2,37 +2,52 @@ package client
 
 import (
 	"context"
+	sdklog "cosmossdk.io/log"
 	"fmt"
-	"math/big"
+	"github.com/dydxprotocol/v4-chain/protocol/daemons/bridge/client/types"
+	"github.com/dydxprotocol/v4-chain/protocol/daemons/bridge/client/types/constants"
+	libtime "github.com/dydxprotocol/v4-chain/protocol/lib/time"
 	"time"
 
 	"github.com/cometbft/cometbft/libs/log"
-	"github.com/cosmos/cosmos-sdk/telemetry"
 	appflags "github.com/dydxprotocol/v4-chain/protocol/app/flags"
 	"github.com/dydxprotocol/v4-chain/protocol/daemons/bridge/api"
-	"github.com/dydxprotocol/v4-chain/protocol/daemons/bridge/client/types"
-	"github.com/dydxprotocol/v4-chain/protocol/daemons/constants"
 	daemonflags "github.com/dydxprotocol/v4-chain/protocol/daemons/flags"
 	daemontypes "github.com/dydxprotocol/v4-chain/protocol/daemons/types"
-	libeth "github.com/dydxprotocol/v4-chain/protocol/lib/eth"
-	"github.com/dydxprotocol/v4-chain/protocol/lib/metrics"
 	bridgetypes "github.com/dydxprotocol/v4-chain/protocol/x/bridge/types"
-	eth "github.com/ethereum/go-ethereum"
-	ethcommon "github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/ethclient"
-	ethrpc "github.com/ethereum/go-ethereum/rpc"
 )
 
+// Client encapsulates the logic and interface for the bridge daemon. The bridge daemon periodically queries the
+// Ethereum blockchain for new bridge events and relays them to the Cosmos gRPC server.
+type Client struct {
+	daemontypes.HealthCheckable
```

Comment:
Add a new `Client` type. Compose this struct with a HealthCheckable field that is initialized below.